pkg/util/kernel/find_headers.go

```diff
@@ -78,7 +78,7 @@ func GetKernelHeaders(headerDirs []string, headerDownloadDir string) ([]string,
 	}
 	log.Debugf("unable to find downloaded kernel headers: %s", err)
 
-	if err = downloadHeaders(headerDownloadDir); err == nil {
+	if err = downloadHeaders(headerDownloadDir, aptConfigDir, yumReposDir, zypperReposDir); err == nil {
```

Comment:
I suggest creating a non-exported type with the config values, and having `downloadHeaders` (and other functions that need the config values) be functions of that type: i.e. `func (h *headerDownloader) downloadHeaders(headerDownloadDir) error`. This way every function that needs these values doesn't add 3 arguments to their function signature.